repair/repair_task.hh

```diff
@@ -80,6 +84,10 @@ public:
             });
         }
     }
+
+    virtual tasks::is_abortable is_abortable() const noexcept override {
+        return tasks::is_abortable(!_abort_subscription);
+    }
```

Comment:
I'm starting to think maybe we should have separate types for top level and sub-tasks. Not for this PR, just thinking out loud.

Reply:
Once node ops will become tasks I'm not sure this will be needed. is_abortable could check parent_id instead in this case.

I though about moving the logic to the task manager api layer to disallow aborting of child tasks but I think it may still be desirable, e.g. for some compaction tasks where you want to abort e.g. major compaction on one table but continue on others if it was initiated for the whole keyspace.